Bugfix #1727 - Price for Recently Viewed Products doesn't update if catalog rule for Customer Group applied it

vladi-j opened the pull request:
Issue: #1727 

If discount is applied to Customer Group, price is changed in recently viewed products after customer logged out.

ejnshtein commented:
Guess, you need to resolve conflicts before this PR can be reviewed and merged.

vladi-j commented:
Resolved merge conflicts. As implementation was changed a lot, I completely removed my code and very slightly adjusted already existing as it was performing the same operation.

ejnshtein commented:
Okay! Hopefully, it will work as expected! Thanks!

AleksandrsKondratjevs commented on packages/scandipwa/src/component/RecentlyViewedWidget/RecentlyViewedWidget.container.js:
```diff
@@ -50,13 +50,12 @@ export class RecentlyViewedWidgetContainer extends PureComponent {
             store
         } = this.props;
 
-        if (shouldBeUpdated && Object.entries(recentProducts).length !== 0) {
```
I assume if we remove this check then it will be fetching products each time user visiting page with Recently Viewed Products widget, which is okey, but as I know placeholders are shown when fetching data from BE only if this 'shouldBeUpdated' is true. It is true when user visiting PDP and product is added into localStorage 'RecentlyViewedProducts'. Can we set bool to be true each time we are fetching data for widget?  So placeholders will appear while data will be updating and there will be no mystery appear, for example with price change.

AleksandrsKondratjevs replied:
Btw, since we fetching data from BE each time user sees widget we can remove logic for setting shouldBeUpdated when adding product into RecentlyViewedProducts in localStorage. Also would be great to rename it, something like 'isLoading' would make more sense.

vladi-j commented:
As requested, 

- setting bool to true each time we are fetching data for widget.
- removed logic for setting shouldBeUpdated after product addition
- renamed bool to 'isLoading'

AleksandrsKondratjevs commented on packages/scandipwa/src/component/RecentlyViewedWidget/RecentlyViewedWidget.container.js:
```diff
@@ -50,13 +53,14 @@ export class RecentlyViewedWidgetContainer extends PureComponent {
 
-        if (shouldBeUpdated && Object.entries(recentProducts).length !== 0) {
+        if (Object.entries(recentProducts).length !== 0) {
+            updateLoadStatus(true);
```
Just to make it more clear, can we set loading status in RecentlyViewedProducts.dispatcher ? I guess it can be set in prepareRequest function.

AleksandrsKondratjevs reviewed (approved):
Minor changes required. Otherwise, seems legit for me.
